snap/revision.go

```diff
@@ -56,6 +56,18 @@ func (r Revision) MarshalJSON() ([]byte, error) {
 	return []byte(`"` + r.String() + `"`), nil
 }
 
+func (r *Revision) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var s string
+	if err := unmarshal(&s); err != nil {
+		return err
+	}
+	return r.UnmarshalJSON([]byte(`"` + s + `"`))
```

Comment:
maybe extract a small helper for the "..." case from UnarmashalJSON instead of adding the "" here? or just call ParseRevision directly, is little code actually